### PR TITLE
Tcb Logs for successful, failure and locked logins.

### PR DESCRIPTION
**Pull request by vivek-opsmx**
We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](https://spinnaker.io/community/contributing/).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](https://spinnaker.io/community/contributing/submitting/).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java [hunk 1]
```diff
@@ -60,6 +61,7 @@ public void onApplicationEvent(AbstractAuthenticationEvent event) {
       } else if (!event.getAuthentication().isAuthenticated()
           && event instanceof AbstractAuthenticationFailureEvent) {
         log.debug("publishEvent AbstractAuthenticationFailureEvent");
+        log.info("Login : User '"+event.getAuthentication().getName()+"' login failed due to incorrect credentials.");
```
Comment (Pranav-b-7) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java:64]:
```suggestion
        log.info("User : {} login failed due to incorrect credentials.", event.getAuthentication().getName());
```
Can we change the log statement like this ?

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java [hunk 2]
```diff
@@ -68,10 +70,12 @@ public void onApplicationEvent(AbstractAuthenticationEvent event) {
             .getName()
             .equals("org.springframework.security.providers.ExpiringUsernameAuthenticationToken")) {
           log.debug("publishEvent LogoutSuccessEvent with ExpiringUsernameAuthenticationToken");
+          log.info("Logout : User '"+event.getAuthentication().getName()+"' logged out.");
```
Comment (Pranav-b-7) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java:73]:
```suggestion
          log.info("User : {} logged out.", event.getAuthentication().getName());
```

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java [hunk 3]
```diff
@@ -68,10 +70,12 @@ public void onApplicationEvent(AbstractAuthenticationEvent event) {
...
           handleAuthenticationEvent(event, AuditEventType.SUCCESSFUL_USER_LOGOUT_AUDIT);
           return;
         }
         log.debug("publishEvent LogoutSuccessEvent");
+        log.info("Logout : User '"+event.getAuthentication().getName()+"' logged out manually.");
```
Comment (Pranav-b-7) [inline gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java:78]:
```suggestion
        log.info("User : {} logged out manually.", event.getAuthentication().getName());
```

**Review by Pranav-b-7 — CHANGES_REQUESTED** [review]

File: gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java [hunk 4]
```diff
@@ -49,6 +49,7 @@ public void onApplicationEvent(AbstractAuthenticationEvent event) {
       if (event.getAuthentication().isAuthenticated()
           && event instanceof InteractiveAuthenticationSuccessEvent) {
         log.debug("publishEvent InteractiveAuthenticationSuccessEvent");
+        log.info("Login: User '"+event.getAuthentication().getName()+"' logged in successfully.");
```
[the remark this answers is not in the document]
Comment (vivek-opsmx) [reply gate-web/src/main/java/com/opsmx/spinnaker/gate/audit/AuthenticationAuditListener.java:52]: this message was suggested By @sagayd  for TCB  customer we are following log info structure like this [action: log message]

**Review by Pranav-b-7 — APPROVED** [review]

**Pull request merged**